### DISCUSSION TITLE
Polycrystal with random euler angle distribution

### DISCUSSION THREAD

**Discussion by ghost** [idaholab/moose#27893] [Q&A Modules: Solid mechanics]
Hi I have a polycrystal mesh file from neper and I've used "ElementPropertyReadFile" to have a random distribution of the euler angles for the ComputeElasticityTensorCP. In paraview I can see different euler angles for each grain but I have the same elasticity tensor for all grains when I output elasticity tensor components as : 
```
[C1111]
    type = RankFourAux
    variable = C1111
    rank_four_tensor = elasticity_tensor
    index_l = 0
    index_j = 0
    index_k = 0
    index_i = 0
    execute_on = timestep_end
```
how do I get the new rotated elasticity tensor

**Comment by GiudGiud** [comment]
@jiangwen84

**Reply by ghost** [reply]
THink this is because we are using a local isotropic elasticity tensor expecting the rotation to give the anisotropy for the polycrystal

**Reply by sapitts** [reply]
Hi @amassaf, in case you haven't already found this class, you can use [ComputeUpdatedEulerAngle](https://mooseframework.inl.gov/source/materials/crystal_plasticity/ComputeUpdatedEulerAngle.html) to output the updated euler angles. However, this information will be similar to what you have seen in paraview.

Because the stress-update based crystal plasticity code is written to refer to the reference configuration, we only calculate the rotation of the Rank-4 elasticity tensor once at the beginning of the simulation. Updates to the rotation matrix are computed in the ComputeMultipleCrystalPlasticityStress class https://github.com/idaholab/moose/blob/next/modules/solid_mechanics/src/materials/crystal_plasticity/ComputeMultipleCrystalPlasticityStress.C#L315-L317

@dewenyushu do you have insights into how to update the Rank-4 elasticity tensor from the `_updated_rotation[_qp]` Rank-2 tensor?
